src/helpers/pagination.helper.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  PaginationMetadata,
+  PaginationParams,
+  DEFAULT_LIMIT,
+  DEFAULT_PAGE,
+} from '../types/pagination.types';
+
+export function calculatePaginationMetadata(
+  page: number,
+  limit: number,
+  total: number,
+): PaginationMetadata {
+  const totalPages = Math.ceil(total / limit);
```

Comment:
Suggestion: Validate that "limit" is greater than zero to prevent division by zero or invalid totalPages calculation.